web/src/settings_muted_topics.js

```diff
@@ -3,52 +3,83 @@ import $ from "jquery";
+    for (const visibility_policy of visibility_policies) {
+        const user_topics_list =
+            user_topics.get_user_topics_for_visibility_policy(visibility_policy);
+        for (const user_topic of user_topics_list) {
+            user_topic.visibility_policy = visibility_policy;
+            all_user_topics.push(user_topic);
+        }
+    }
+    const $user_topics_table = $("#user_topics_table");
+    const $search_input = $("#user_topics_search");
+
+    ListWidget.create($user_topics_table, all_user_topics, {
+        name: "user-topics-list",
         modifier(user_topic) {
-            return render_user_topic_ui_row({user_topic});
+            const context = {
+                user_topic,
+                development_environment: page_params.development_environment,
+                user_topic_visibility_policy_values:
+                    settings_config.user_topic_visibility_policy_values,
+            };
+            return render_user_topic_ui_row(context);
         },
         filter: {
             $element: $search_input,
             predicate(item, value) {
                 return item.topic.toLocaleLowerCase().includes(value);
             },
             onupdate() {
-                ui.reset_scrollbar($muted_topics_table.closest(".progressive-table-wrapper"));
+                ui.reset_scrollbar($user_topics_table.closest(".progressive-table-wrapper"));
             },
         },
-        $parent_container: $("#muted-topic-settings"),
-        $simplebar_container: $("#muted-topic-settings .progressive-table-wrapper"),
+        init_sort: ["numeric", "date_updated"],
+        initially_descending_sort: true,
+        $parent_container: $("#user-topic-settings"),
+        $simplebar_container: $("#user-topic-settings .progressive-table-wrapper"),
     });
 }
 
 export function set_up() {
     loaded = true;
-    $("body").on("click", ".settings-unmute-topic", function (e) {
-        const $row = $(this).closest("tr");
-        const stream_id = Number.parseInt($row.attr("data-stream-id"), 10);
-        const topic = $row.attr("data-topic");
-
-        e.stopPropagation();
-
-        user_topics.set_user_topic_visibility_policy(
-            stream_id,
-            topic,
-            user_topics.all_visibility_policies.INHERIT,
-        );
-    });
+
+    if (page_params.development_environment) {
+        $("body").on("change", ".settings_user_topic_visibility_policy", function (e) {
+            const $row = $(this).closest("tr");
+            const stream_id = Number.parseInt($row.attr("data-stream-id"), 10);
+            const topic = $row.attr("data-topic");
+            const visibility_policy = this.value;
+
+            e.stopPropagation();
+
+            user_topics.set_user_topic_visibility_policy(stream_id, topic, visibility_policy);
+        });
+    } else {
+        $("body").on("click", ".settings-unmute-topic", function (e) {
```

Comment:
Should `settings-unmute-topic` be renamed to `settings-remove-user-topic-visibility-policy`?